app/src/main/java/com/github/se/cyrcle/ui/profile/CreateProfileScreen.kt

```diff
@@ -28,29 +29,37 @@ fun CreateProfileScreen(navigationActions: NavigationActions, userViewModel: Use
   val defaultUser = User(UserPublic("", ""), UserDetails())
 
   val validationToastText = stringResource(R.string.create_profile_validation_toast)
+  val welcomeBonusText = stringResource(R.string.welcome_bonus_toast)
+  val combinedToastText = "$validationToastText\n$welcomeBonusText"
+
   val errorToastText = stringResource(R.string.create_profile_error_toast)
   val accountExistsToastText = stringResource(R.string.create_profile_account_already_exists)
 
   val onSignInClick = { createdUser: User ->
     userViewModel.authenticate(
         // On successful authentication
         { userId ->
-          val user = createdUser.copy(public = createdUser.public.copy(userId = userId))
+          val userWithId = createdUser.copy(public = createdUser.public.copy(userId = userId))
 
           // Check if user already exists
-          userViewModel.doesUserExist(user) { userExists ->
+          userViewModel.doesUserExist(userWithId) { userExists ->
             if (userExists) {
               Toast.makeText(context, accountExistsToastText, Toast.LENGTH_SHORT).show()
               navigationActions.goBack()
             } else {
-
+              val userWithCoins =
+                  userWithId.copy(
+                      details =
+                          userWithId.details?.copy(
+                              wallet = Wallet(100) // Start with 100 coins
```

Comment:
Same here, use a constant. 
(We could also have the default constructor automatically credit the default number of coins, and take no parameters. to prevent creating wallet with different balances)